diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 4ff90f6..e4e79f4 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -124,11 +124,24 @@ void register_irq_keyhandler(unsigned char key, irq_keyhandler_fn_t fn,
     key_table[key].diagnostic = diagnostic;
 }
 
+void new_global(void)
+{
+    WARN();
+}
+
+static void noinline new_static(void)
+{
+    get_random();
+}
+
 static void show_handlers(unsigned char key)
 {
     unsigned int i;
 
     printk("'%c' pressed -> showing installed handlers\n", key);
+
+    new_static();
+
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ )
         if ( key_table[i].fn )
             printk(" key '%c' (ascii '%02x') => %s\n",
diff --git a/xen/common/random.c b/xen/common/random.c
index fb805b0..79ece82 100644
--- a/xen/common/random.c
+++ b/xen/common/random.c
@@ -8,6 +8,8 @@
 static DEFINE_PER_CPU(unsigned int, seed);
 unsigned int __read_mostly boot_random;
 
+void new_global(void);
+
 unsigned int get_random(void)
 {
     unsigned int next = this_cpu(seed), val = arch_get_random();
@@ -28,6 +30,8 @@ unsigned int get_random(void)
 
     this_cpu(seed) = next;
 
+    new_global();
+
     return val;
 }
 
