diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index e4e79f4..77ee4fb 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -126,11 +126,13 @@ void register_irq_keyhandler(unsigned char key, irq_keyhandler_fn_t fn,
 
 void new_global(void)
 {
+    printk("patching new_global\n");
     WARN();
 }
 
 static void noinline new_static(void)
 {
+    printk("patching new_static\n");
     get_random();
 }
 
