diff --git a/xen/arch/x86/acpi/cpu_idle.c b/xen/arch/x86/acpi/cpu_idle.c
index a21aeed..a1eecec 100644
--- a/xen/arch/x86/acpi/cpu_idle.c
+++ b/xen/arch/x86/acpi/cpu_idle.c
@@ -329,9 +329,14 @@ static void dump_cx(unsigned char key)
     unsigned int cpu;
 
     printk("'%c' pressed -> printing ACPI Cx structures\n", key);
-    for_each_online_cpu ( cpu )
+
+    for_each_online_cpu ( cpu ) {
+        printk("got herex %u\n", cpu);
         if (processor_powers[cpu])
             print_acpi_power(cpu, processor_powers[cpu]);
+    }
+
+    printk("got here\n");
 }
 
 static int __init cpu_idle_key_init(void)
