diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 4ff90f6..487dcd1 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -126,9 +126,13 @@ void register_irq_keyhandler(unsigned char key, irq_keyhandler_fn_t fn,
 
 static void show_handlers(unsigned char key)
 {
-    unsigned int i;
+    unsigned int i, ret;
+
+    printk("'%c' wrmsr pressed -> showing installed handlers\n", key);
+
+    ret = wrmsr_safe(0xFFAABB, 12344);
+    printk("ret = %d\n", ret);
 
-    printk("'%c' pressed -> showing installed handlers\n", key);
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ )
         if ( key_table[i].fn )
             printk(" key '%c' (ascii '%02x') => %s\n",
