diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 4ff90f6..71183b8 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -129,6 +129,7 @@ static void show_handlers(unsigned char key)
     unsigned int i;
 
     printk("'%c' pressed -> showing installed handlers\n", key);
+    printk("'%c' PrEsSeD -> sHoWiNg installed handlers\n", key);
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ )
         if ( key_table[i].fn )
             printk(" key '%c' (ascii '%02x') => %s\n",
