diff --git a/xen/arch/x86/time.c b/xen/arch/x86/time.c
index 687e39b..061c72e 100644
--- a/xen/arch/x86/time.c
+++ b/xen/arch/x86/time.c
@@ -1921,12 +1921,18 @@ void tsc_set_info(struct domain *d,
     }
 }
 
+extern int global_x;
+extern int global_z;
+
 /* vtsc may incur measurable performance degradation, diagnose with this */
 static void dump_softtsc(unsigned char key)
 {
     struct domain *d;
     int domcnt = 0;
 
+    printk("global_x = %d\n", global_x);
+    printk("global_z = %d\n", global_z);
+
     tsc_check_reliability();
     if ( boot_cpu_has(X86_FEATURE_TSC_RELIABLE) )
         printk("TSC marked as reliable, "
diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 4ff90f6..873911f 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -124,11 +124,47 @@ void register_irq_keyhandler(unsigned char key, irq_keyhandler_fn_t fn,
     key_table[key].diagnostic = diagnostic;
 }
 
+int global_x;
+char global_y[8192];
+int global_z = 42;
+static int static_x;
+static char static_y[8192];
+static int static_z = 43;
+
 static void show_handlers(unsigned char key)
 {
     unsigned int i;
+    int found = 0;
 
     printk("'%c' pressed -> showing installed handlers\n", key);
+
+    printk("global_x = %d\n", global_x);
+    printk("global_z = %d\n", global_z);
+    printk("static_x = %d\n", static_x);
+    printk("static_z = %d\n", static_z);
+
+    for ( i = 0; i < 8192; i++ )
+    {
+        if ( global_y[i] != 0 )
+            found = 1;
+    }
+    if ( found )
+        printk("global_y != 0\n");
+
+    found = 0;
+    for ( i = 0; i < 8192; i++ )
+    {
+        if ( static_y[i] != 0 )
+            found = 1;
+    }
+    if ( found )
+        printk("static_y != 0\n");
+
+    global_x++;
+    global_z++;
+    static_x++;
+    static_z++;
+
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ )
         if ( key_table[i].fn )
             printk(" key '%c' (ascii '%02x') => %s\n",
