diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 873911f..654a1ea 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -160,10 +160,10 @@ static void show_handlers(unsigned char key)
     if ( found )
         printk("static_y != 0\n");
 
-    global_x++;
-    global_z++;
-    static_x++;
-    static_z++;
+    global_x -= 7;
+    global_z -= 7;
+    static_x -= 7;
+    static_z -= 7;
 
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ )
         if ( key_table[i].fn )
