diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 4ff90f6..160b755 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -128,7 +128,10 @@ static void show_handlers(unsigned char key)
 {
     unsigned int i;
 
-    printk("'%c' pressed -> showing installed handlers\n", key);
+    printk("'%c' bug pressed -> showing installed handlers\n", key);
+
+    WARN();
+
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ )
         if ( key_table[i].fn )
             printk(" key '%c' (ascii '%02x') => %s\n",
