diff --git a/xen/common/keyhandler.c b/xen/common/keyhandler.c
index 5d21e48..6d0c1bd 100644
--- a/xen/common/keyhandler.c
+++ b/xen/common/keyhandler.c
@@ -63,7 +63,10 @@ void register_keyhandler(unsigned char key, struct keyhandler *handler)
 static void show_handlers(unsigned char key)
 {
     int i;
-    printk("'%c' pressed -> showing installed handlers\n", key);
+    printk("'%c' bug pressed -> showing installed handlers\n", key);
+
+    WARN();
+
     for ( i = 0; i < ARRAY_SIZE(key_table); i++ ) 
         if ( key_table[i] != NULL ) 
             printk(" key '%c' (ascii '%02x') => %s\n", 
